api/tests/mocha/services/generate-xform.spec.js

```diff
@@ -54,21 +57,21 @@ describe('generate-xform service', () => {
         on: sinon.stub()
       };
       sinon.stub(childProcess, 'spawn').returns(spawned);
-      return setup(dirname).then(given => {
-        const generate = service._generate(given.xform);
+      return setup(dirname).then(files => {
+        const generate = service._generate(files.xform);
         if (err) {
           spawned.stderr.on.args[0][1](err);
           spawned.on.args[0][1](100);
         } else {
           // child process outputs then closes with code 0
-          spawned.stdout.on.args[0][1](given.form);
+          spawned.stdout.on.args[0][1](files.givenForm);
           spawned.on.args[0][1](0);
-          spawned.stdout.on.args[1][1](given.model);
+          spawned.stdout.on.args[1][1](files.givenModel);
           spawned.on.args[2][1](0);
         }
         return generate.then(actual => {
-          expect(actual.form).to.equal(given.form);
-          expect(actual.model).to.equal(given.model);
+          expect(actual.form).to.equal(files.expectedForm);
```

Comment:
I find that this pattern of unit testing is easy to over-constrain the system. You might consider keeping one or two "runTest" scenarios, but making the tests like "multimedia" more targeted.

Reply:
Yeah I'm not really happy with the current testing of this unit. I'd like to find a decent way to integration test it and then make the unit tests more lightweight. Mind if I think on this and submit it in another PR later?

Reply:
definitely fine with that